oracle-plugin/src/e2e-test/features/source/OracleRunTimeMacro.feature

```diff
@@ -0,0 +1,307 @@
+    Then Click on the Macro button of Property: "host" and set the value to: "oracleHost"
+    Then Click on the Macro button of Property: "port" and set the value to: "oraclePort"
+    Then Click on the Macro button of Property: "user" and set the value to: "oracleUser"
+    Then Click on the Macro button of Property: "password" and set the value to: "oraclePassword"
+    Then Click on the Macro button of Property: "database" and set the value to: "oracleDatabase"
+    Then Select radio button plugin property: "connectionType" with value: "service"
+    Then Select radio button plugin property: "role" with value: "sysdba"
+    Then Enter input plugin property: "referenceName" with value: "sourceRef"
+    Then Enter textarea plugin property: "importQuery" with value: "selectQuery"
+    Then Validate "Oracle" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "Oracle2"
+    Then Select dropdown plugin property: "select-jdbcPluginName" with option value: "driverName"
+    Then Replace input plugin property: "host" with value: "host" for Credentials and Authorization related fields
+    Then Replace input plugin property: "port" with value: "port" for Credentials and Authorization related fields
+    Then Replace input plugin property: "database" with value: "databaseName"
+    Then Replace input plugin property: "tableName" with value: "targetTable"
+    Then Replace input plugin property: "dbSchemaName" with value: "schema"
+    Then Replace input plugin property: "user" with value: "username" for Credentials and Authorization related fields
+    Then Replace input plugin property: "password" with value: "password" for Credentials and Authorization related fields
+    Then Enter input plugin property: "referenceName" with value: "targetRef"
+    Then Select radio button plugin property: "connectionType" with value: "service"
+    Then Select radio button plugin property: "role" with value: "sysdba"
+    Then Validate "Oracle2" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "invalidDriverName" for key "oracleDriverName"
+    Then Enter runtime argument value "invalidHost" for key "oracleHost"
+    Then Enter runtime argument value "invalidPort" for key "oraclePort"
+    Then Enter runtime argument value "invalidUserName" for key "oracleUser"
+    Then Enter runtime argument value "invalidPassword" for key "oraclePassword"
+    Then Enter runtime argument value "invalidDatabaseName" for key "oracleDatabase"
+    Then Run the preview of pipeline with runtime arguments
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "failed"
+    Then Close the pipeline logs
+    Then Close the preview
+
+  @ORACLE_SOURCE_TEST @ORACLE_SINK_TEST @Oracle_Required
```

Comment:
Why are we using @Oracle_Required hook in our test cases. This concept is related to Github Actions when we only run some specific tests whenever a PR is raised to decrease the Build run time. Is it done in any of the tests written by google team ?